### PR TITLE
feat(core): extract js related code from affected and hasher

### PR DESCRIPTION
**Pull request by meeroslav**
- [X] Extract JS related code from `affected` graph
- [x] Extract JS related code from `hasher`
- [x] Add deep imports to barrel export 

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by mandarini — APPROVED** [review]
Storybook, Vite and React changes look good to me!

**Review by Coly010 — APPROVED** [review]

**Review by nartc — APPROVED** [review]

**Review by jaysoo — APPROVED** [review]

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/nx/src/plugins/js/index.ts [hunk 1]
```diff
@@ -16,7 +16,23 @@ import { projectGraphCacheDirectory } from '../../utils/cache-directory';
...
+ */
+export {
+  findNodes,
+  getRootTsConfigPath,
+  getRootTsConfigFileName,
+  readTsConfig,
+  resolveModuleByImport,
+} from './utils/typescript';
+export { getLockFileName, createLockFile } from './lock-file/lock-file';
+export { createPackageJson } from './package-json/create-package-json';
```
[the remark this answers is not in the document]
Comment (meeroslav) [reply packages/nx/src/plugins/js/index.ts:32]: Exactly. Most likely entire `plugins/js` should be in `@nrwl/js`

File: packages/nx/src/plugins/js/index.ts [hunk 2]
```diff
@@ -16,7 +16,23 @@ import { projectGraphCacheDirectory } from '../../utils/cache-directory';
...
+export { TargetProjectLocator } from './project-graph/build-dependencies/target-project-locator';
+export { getTouchedProjects } from './project-graph/affected/touched-projects';
+export { hashTsConfig } from './hasher/hasher';
```
[the remark this answers is not in the document]
Comment (meeroslav) [reply packages/nx/src/plugins/js/index.ts:35]: Where would you export it from? We discussed having a barrel export and a [linter rule ](https://github.com/nrwl/nx/pull/16253)that prevents deep imports to enable easier extraction of this functionality from the `nx` package in the near future.

**Review by barbados-clemens — APPROVED** [review]

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: e2e/jest/src/jest.test.ts [hunk 3]
```diff
@@ -48,7 +48,7 @@ describe('Jest', () => {
     updateFile(
       `libs/${mylib}/setup.ts`,
       stripIndents`
-      const { registerTsProject } = require('nx/src/utils/register');
+      const { registerTsProject } = require('nx/src/plugins/js/utils/register');
```
Comment (FrozenPandaz) [inline e2e/jest/src/jest.test.ts:51]: followup: This should use `@nrwl/js/src/internals`

File: packages/linter/src/generators/init/init.ts [hunk 4]
```diff
@@ -10,9 +10,9 @@ import {
 import {
   eslintConfigPrettierVersion,
   eslintVersion,
-  nxVersion,
   typescriptESLintVersion,
 } from '../../utils/versions';
+import { nxVersion } from '@nx/js/src/utils/versions';
```
Comment (FrozenPandaz) [inline packages/linter/src/generators/init/init.ts:15]: Why did this change?

File: packages/linter/src/generators/workspace-rules-project/workspace-rules-project.ts [hunk 5]
```diff
@@ -16,8 +16,11 @@ import {
 import { getRelativePathToRootTsConfig } from '@nx/js';
 import { join } from 'path';
 import { workspaceLintPluginDir } from '../../utils/workspace-lint-rules';
-import { swcCoreVersion, swcNodeVersion } from 'nx/src/utils/versions';
-import { nxVersion } from '../../utils/versions';
+import {
+  nxVersion,
```
Comment (FrozenPandaz) [inline packages/linter/src/generators/workspace-rules-project/workspace-rules-project.ts:20]: What's wrong with using the one from this package?

File: packages/linter/src/generators/workspace-rules-project/workspace-rules-project.ts [hunk 6]
```diff
@@ -16,8 +16,11 @@ import {
...
+  swcCoreVersion,
+  swcNodeVersion,
```
Comment (FrozenPandaz) [inline packages/linter/src/generators/workspace-rules-project/workspace-rules-project.ts:22]: I don't think we should import the versions from `@nx/js`, I think we should import a function like `{ addSwcDependencies } from '@nx/js/src/utils/add-swc';`

File: packages/linter/src/migrations/update-14-1-9/add-swc-deps-if-needed.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import { addDependenciesToPackageJson, formatFiles, Tree } from '@nrwl/devkit';
-import { swcCoreVersion, swcNodeVersion } from 'nx/src/utils/versions';
+import { swcCoreVersion, swcNodeVersion } from '../../utils/versions';
 import { WORKSPACE_PLUGIN_DIR } from '../../generators/workspace-rules-project/workspace-rules-project';
 
 export default async function addSwcNodeIfNeeded(tree: Tree) {
```
[the remark this answers is not in the document]
Comment (FrozenPandaz) [reply packages/linter/src/migrations/update-14-1-9/add-swc-deps-if-needed.ts:5]: This was not addressed.

File: packages/linter/src/utils/versions.ts [hunk 8]
```diff
@@ -1,7 +1,4 @@
-export const nxVersion = require('../../package.json').version;
```
Comment (FrozenPandaz) [inline packages/linter/src/utils/versions.ts:1]: Why remove this?

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/linter/src/utils/versions.ts (hunk 8 above)
Comment (meeroslav) [reply packages/linter/src/utils/versions.ts:1]: I thought it makes sense to have `nx` as a central source of truth when it comes to `nxVersion`.

I'll revert that.

File: packages/linter/src/generators/init/init.ts (hunk 4 above)
Comment (meeroslav) [reply packages/linter/src/generators/init/init.ts:15]: I thought it made sense to have nx as a central source of truth regarding nxVersion.

I'll revert that.

File: packages/linter/src/generators/workspace-rules-project/workspace-rules-project.ts (hunk 6 above)
Comment (meeroslav) [reply packages/linter/src/generators/workspace-rules-project/workspace-rules-project.ts:22]: This got lost on rebase; I'll push those changes again.

File: packages/nx-plugin/src/migrations/update-14-1-9/add-swc-deps-if-needed.ts [hunk 9]
```diff
@@ -4,7 +4,7 @@ import {
   installPackagesTask,
   Tree,
 } from '@nrwl/devkit';
-import { swcCoreVersion, swcNodeVersion } from 'nx/src/utils/versions';
+import { swcCoreVersion, swcNodeVersion } from '../../utils/versions';
 
 export default async function addSwcNodeIfNeeded(tree: Tree) {
   addDependenciesToPackageJson(
```
[the remark this answers is not in the document]
Comment (meeroslav) [reply packages/nx-plugin/src/migrations/update-14-1-9/add-swc-deps-if-needed.ts:10]: This got lost on rebase; I'll push those changes again.

File: packages/linter/src/migrations/update-14-1-9/add-swc-deps-if-needed.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (meeroslav) [reply packages/linter/src/migrations/update-14-1-9/add-swc-deps-if-needed.ts:5]: This got lost on rebase; I'll push those changes again.

File: e2e/jest/src/jest.test.ts (hunk 3 above)
Comment (meeroslav) [reply e2e/jest/src/jest.test.ts]: Something to discuss tomorrow...

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**